deisctl/config/api.go

```diff
@@ -1,8 +0,0 @@
-package config
```

Comment:
@iancoffey I'm not sure about relocating this interface to the `etcdclient` package.  This interface looks like / was previously a generic "config client" interface of which many implementations are possible-- your new mock implementation of that being one of them.  But moving this interface to the `etcdclient` means users of the interface will be coding to an abstraction (the interface) whose package name is polluted with the name of a specific underlying implementation.

Was there a specific reason for moving it?